File: ProjetSopraFinal/src/main/java/com/example/demo/model/Stagiaire.java
package com.example.demo.model;

import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;

import com.example.demo.jsonview.JsonViews;
import com.fasterxml.jackson.annotation.JsonView;


@Entity
@DiscriminatorValue("S")
public class Stagiaire extends RessourceHumaine {
	
	private Ordinateur ordinateur;
	private Formation formation;

	public Stagiaire() {
		super();
	}

	@JsonView(JsonViews.Common.class)
	@OneToOne
	public Ordinateur getOrdinateur() {
		return ordinateur;
	}

	public void setOrdinateur(Ordinateur ordinateur) {
		this.ordinateur = ordinateur;
	}

	@JsonView(JsonViews.Common.class)
	@ManyToOne
	@JoinColumn (name = "formation")
	public Formation getFormation() {
		return formation;
	}

	public void setFormation(Formation formation) {
		this.formation = formation;
	}
	
	

}

File: AngularSopraFinal/src/app/model/stagiaire.ts
import {Ordinateur} from './ordinateur';
import {Formation} from './formation';
import {RessourceHumaine} from './ressourceHumaine';

export class Stagiaire  extends RessourceHumaine{

  constructor(private _ordinateur?, private _formation?) {
    super('stagiaire');
  }

  public get ordinateur(): Ordinateur {
    return this._ordinateur
  }
  public set ordinateur(_ordinateur: Ordinateur) {
    this._ordinateur = _ordinateur;
  }

  public get formation(): Formation {
    return this._formation
  }
  public set formation(_formation: Formation) {
    this._formation = _formation;
  }
}